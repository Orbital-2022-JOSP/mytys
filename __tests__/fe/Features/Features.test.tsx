import { render } from '@testing-library/react';
import { Features } from '../../../components/Features/Features';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The Features Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<Features />);

            expect(container).toMatchSnapshot()
        })
})

