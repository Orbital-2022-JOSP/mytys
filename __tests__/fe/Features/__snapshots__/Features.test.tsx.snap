// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Features Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <div
      class="max-w-xl mb-10 md:mx-auto sm:text-center lg:max-w-2xl md:mb-12"
    >
      <h2
        class="max-w-lg mb-6 font-sans text-3xl font-bold leading-none tracking-tight text-gray-900 sm:text-4xl md:mx-auto"
      >
        <span
          class="relative inline-block"
        >
          <svg
            class="absolute top-0 left-0 z-0 hidden w-32 -mt-8 -ml-20 text-light-blue-gray-100 lg:w-32 lg:-ml-28 lg:-mt-10 sm:block"
            fill="currentColor"
            viewBox="0 0 42 21"
          >
            <defs>
              <pattern
                height=".30"
                id="27df4f81-c854-45de-942a-fe90f7a300f9"
                width=".135"
                x="0"
                y="0"
              >
                <circle
                  cx="1"
                  cy="1"
                  r=".7"
                />
              </pattern>
            </defs>
            <rect
              fill="url(#27df4f81-c854-45de-942a-fe90f7a300f9)"
              height="24"
              width="52"
            />
          </svg>
          <span
            class="relative"
          />
        </span>
         
        <span
          class="text-light-blue-500"
        >
          MYTYS
        </span>
      </h2>
      <p
        class="text-base text-gray-700 md:text-lg"
      >
        The features our webpage offers
      </p>
    </div>
    <div
      class="grid max-w-screen-lg gap-8 row-gap-10 mx-auto lg:grid-cols-2"
    >
      <div
        class="flex flex-col max-w-md sm:mx-auto sm:flex-row"
      >
        <div
          class="mr-4"
        >
          <div
            class="flex items-center justify-center w-12 h-12 mb-4 rounded-full bg-indigo-50"
          >
            <svg
              class="w-10 h-10 text-light-blue-accent-400"
              stroke="currentColor"
              viewBox="0 0 52 52"
            >
              <polygon
                fill="none"
                points="29 13 14 29 25 29 23 39 38 23 27 23"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
            </svg>
          </div>
        </div>
        <div>
          <h6
            class="mb-3 text-xl font-bold leading-5"
          >
            Gamified learning experience
          </h6>
          <p
            class="mb-3 text-sm text-gray-900"
          >
            Our gamified answering system provides a more interactive way for students to learn, and has been shown to cultivate interest and joy while learning
          </p>
          <a
            aria-label=""
            class="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-accent-400 hover:text-light-blue-800"
            href="/"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        class="flex flex-col max-w-md sm:mx-auto sm:flex-row"
      >
        <div
          class="mr-4"
        >
          <div
            class="flex items-center justify-center w-12 h-12 mb-4 rounded-full bg-indigo-50"
          >
            <svg
              class="w-10 h-10 text-light-blue-accent-400"
              stroke="currentColor"
              viewBox="0 0 52 52"
            >
              <polygon
                fill="none"
                points="29 13 14 29 25 29 23 39 38 23 27 23"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
            </svg>
          </div>
        </div>
        <div>
          <h6
            class="mb-3 text-xl font-bold leading-5"
          >
            Large library of practice questions
          </h6>
          <p
            class="mb-3 text-sm text-gray-900"
          >
            We have collated a library of numerous questions from past year papers of the subjects we offer, and we have recieved feedback that these questions were effective in preparing students for their examinations
          </p>
          <a
            aria-label=""
            class="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-accent-400 hover:text-light-blue-800"
            href="/"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        class="flex flex-col max-w-md sm:mx-auto sm:flex-row"
      >
        <div
          class="mr-4"
        >
          <div
            class="flex items-center justify-center w-12 h-12 mb-4 rounded-full bg-indigo-50"
          >
            <svg
              class="w-10 h-10 text-light-blue-accent-400"
              stroke="currentColor"
              viewBox="0 0 52 52"
            >
              <polygon
                fill="none"
                points="29 13 14 29 25 29 23 39 38 23 27 23"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
            </svg>
          </div>
        </div>
        <div>
          <h6
            class="mb-3 text-xl font-bold leading-5"
          >
            Leaderboard and Points system
          </h6>
          <p
            class="mb-3 text-sm text-gray-900"
          >
            Points are given to students who regularly participate in quizzes and who do well in their quizzes. These points will be collated onto a common leaderboard, where other students can view the top performing students for that time period.
          </p>
          <a
            aria-label=""
            class="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-accent-400 hover:text-light-blue-800"
            href="/"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        class="flex flex-col max-w-md sm:mx-auto sm:flex-row"
      >
        <div
          class="mr-4"
        >
          <div
            class="flex items-center justify-center w-12 h-12 mb-4 rounded-full bg-indigo-50"
          >
            <svg
              class="w-10 h-10 text-light-blue-accent-400"
              stroke="currentColor"
              viewBox="0 0 52 52"
            >
              <polygon
                fill="none"
                points="29 13 14 29 25 29 23 39 38 23 27 23"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
            </svg>
          </div>
        </div>
        <div>
          <h6
            class="mb-3 text-xl font-bold leading-5"
          >
            A 
            <span
              class="text-light-blue-500"
            >
              FREE
            </span>
             tuition experience
          </h6>
          <p
            class="mb-3 text-sm text-gray-900"
          >
            As a relatively new project, we are still monitoring our site's traffic and participation. Until further notice, this site's features and services remain free to all users
          </p>
          <a
            aria-label=""
            class="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-accent-400 hover:text-light-blue-800"
            href="/"
          >
            Learn more
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
